packages/compass-connections/src/stores/connections-store-redux.ts

```diff
@@ -0,0 +1,2078 @@
+    const dataService = DataServiceForConnection.get(connectionId);
+
+    void Promise.all([
+      connectionAttempt?.cancelConnectionAttempt(),
+      dataService?.disconnect(),
+    ]).then(
+      () => {
+        debug('connection closed', connectionId);
+      },
+      (err) => {
+        log.error(
+          mongoLogId(1_001_000_314),
+          'Connection UI',
+          'Disconnect attempt failed',
+          { error: (err as Error).message }
+        );
+      }
+    );
+
+    ConnectionAttemptForConnection.delete(connectionId);
+    DataServiceForConnection.delete(connectionId);
+
+    connectionsEventEmitter.emit('disconnected', connectionId, connectionInfo!);
+  };
+};
+
+export const disconnect = (
+  connectionId: ConnectionId
+): ConnectionsThunkAction<void> => {
+  return (dispatch, getState, { logger: { debug } }) => {
+    debug('closing connection with connectionId', connectionId);
+
+    dispatch(cleanupConnection(connectionId));
+
+    dispatch({ type: ActionTypes.Disconnect, connectionId });
+  };
+};
+
+export const removeConnection = (
+  connectionId: ConnectionId
+): ConnectionsThunkAction<void, RemoveConnectionAction> => {
+  return (
+    dispatch,
+    getState,
+    { connectionStorage, track, logger: { debug } }
+  ) => {
+    const connectionInfo = getCurrentConnectionInfo(getState(), connectionId);
+
+    if (!connectionInfo) {
+      return;
+    }
+
+    dispatch(cleanupConnection(connectionInfo.id));
+
+    void connectionStorage.delete?.({ id: connectionId }).catch((err) => {
+      debug('failed to delete connection', err);
+    });
+
+    dispatch({
+      type: ActionTypes.RemoveConnection,
+      connectionId,
+    });
+
+    track('Connection Removed', {}, connectionInfo);
+  };
+};
+
+export const toggleConnectionFavoritedStatus = (
+  connectionId: ConnectionId
+): ConnectionsThunkAction<void, ToggleFavoriteConnectionAction> => {
+  return (dispatch, getState) => {
+    if (isAutoconnectInfo(getState(), connectionId)) {
+      return;
+    }
+
+    if (!hasConnectionForId(getState(), connectionId)) {
+      return;
+    }
+
+    dispatch({ type: ActionTypes.ToggleFavoriteConnection, connectionId });
+
+    // After ToggleFavoriteConnection was dispatched, connectionInfo in state
+    // was already updated to toggle the value, we can use it now to save the
+    // connection in storage
+    const newConnectionInfo = getCurrentConnectionInfo(
+      getState(),
+      connectionId
+    );
+
+    // Making TS happy, should never end up here
+    if (!newConnectionInfo) {
+      throw new Error('No connection info to save');
+    }
+
+    void dispatch(saveConnectionInfo(newConnectionInfo));
+  };
+};
+
+function isRecentConnection(connection: ConnectionState) {
+  return (
+    !connection.info.savedConnectionType ||
+    connection.info.savedConnectionType === 'recent'
+  );
+}
+
+export const removeAllRecentConnections = (): ConnectionsThunkAction<
```

Comment:
Is this still used by single connections? Because I don't think we have UI for it anymore.

Reply:
Yep, only single connections. I wonder if I should just have this in legacy-connections component calling `removeConnection` in a loop instead of an action in a store, probably easier to clean-up later that way 🤔